### PR TITLE
Implement takeAll

### PR DESCRIPTION
**Pull request by xdanny**
Implements takeAll method. 

Resolves: [#8](https://github.com/scalaz/scalaz-ioqueue/issues/8)

### CODE REVIEW

File: src/main/scala/scalaz/ioqueue/Queue.scala [hunk 1]
```diff
@@ -55,6 +55,16 @@ class Queue[A] private (capacity: Int, ref: Ref[State[A]]) {
     Promise.bracket[Nothing, State[A], Unit, Boolean](ref)(acquire)(release)
   }
 
+  /**
+   * Removes all the values in the queue and returns the list of the values. If the queue
+   * is empty returns empty list.
+   */
+  final def takeAll: IO[Nothing, List[A]] =
+    ref.modify[List[A]] {
+      case Surplus(values, putters) => (values.view.toList, Surplus(IQueue.empty[A], putters))
```
Comment (wi101) [inline src/main/scala/scalaz/ioqueue/Queue.scala]: `values.toList` provides the same result as `values.view.toList`

File: src/test/scala/scalaz/ioqueue/QueueSpec.scala [hunk 2]
```diff
@@ -142,6 +147,25 @@ class QueueSpec(implicit ee: ExecutionEnv) extends AbstractRTSSpec with AroundTi
...
+
+  def e13 = unsafeRun(
+    for {
+      queue <- Queue.unbounded[Int]
+      _     <- queue.offer(1)
+      _     <- queue.take
+      v     <- queue.takeAll
+    } yield (v must_=== List.empty)
+  )
+
```
Comment (LukaszByczynski) [inline src/test/scala/scalaz/ioqueue/QueueSpec.scala]: Add test for initial empty Queue

File: src/main/scala/scalaz/ioqueue/Queue.scala (hunk 1 above)
Comment (xdanny) [reply src/main/scala/scalaz/ioqueue/Queue.scala]: Yes that is correct, no need to have view on collection. Will change.

**Comment by wi101** [comment]
@xdanny Good job :)

**Comment by xdanny** [comment]
@wi101 Thanks for the guiding! If @LukaszByczynski has a second review maybe it can be merged?
